EIPS/eip-7069.md

```diff
@@ -0,0 +1,145 @@
+## Specification
+
+| Name | Value | Comment |
+|------|-------|---------|
+| WARM_STORAGE_READ_COST | 100 | From [EIP-2929](./eip-2929.md) |
+| COLD_ACCOUNT_ACCESS | 2600 | From [EIP-2929](./eip-2929.md) |
+| ACCOUNT_CREATION_COST | 25000 | |
+| MIN_RETAINED_GAS | 5000 | |
+| MIN_CALLEE_GAS | 2300 | |
+
+We introduce three new instructions:
+
+- `CALL2` (`0xf8`) with arguments `(target_address, input_offset, input_size, value)` 
+- `DELEGATECALL2` (`0xf9`) with arguments `(target_address, input_offset, input_size)`
+- `STATICCALL2` (`0xfb`) with arguments `(target_address, input_offset, input_size)`
+
+Execution semantics:
+
+1. Charge `WARM_STORAGE_READ_COST` (100) gas.
+2. Pop required arguments from stack, fail with error on stack underflow.
+4. If `value` is non-zero:
+    3a. Fail with error if the current frame is in `static-mode`.
+    3b. Fail with error if the balance of the current account is less than `value`.
+4. Peform (and charge for) memory expansion using `[input_offset, input_size]`.
+5. If `target_address` is not in the `warm_account_list`, charge `COLD_ACCOUNT_ACCESS - WARM_STORAGE_READ_COST` (2500) gas.
+6. If `target_address` is not in the state and the call configuration would result in account creation, charge `ACCOUNT_CREATION_COST` (25000) gas.
+    - The only such case in this EIP is if `value` is non-zero.
+7. Calculate the gas available to callee as caller's remaining gas reduced by `max(ceil(gas/64), MIN_RETAINED_GAS)` (`MIN_RETAINED_GAS` is 5000).
+8. Fail with error if the gas available to callee at this point is less than `MIN_CALLEE_GAS` (2300).
+9. Perform the call with the available gas and configuration.
+10. Push a status code on the stack:
+    11a. `0` if the call was successful.
+    11b. `1` if the call has reverted.
+    11c. `2` if the call has failed.
+11. Gas not used by the callee is returned to the caller.
+
+Note: Unlike `CALL` there is no extra charge for value bearing calls.
+
+*TODO:* Clarify which side (caller/callee) is gas deducted from and where an error originates from.
+
+*TODO:* Mention gas refunds?
+
+*TODO:* Consider option where non-calldata value transfer is not allowed, but there's a specific `TRANSFER`/`PAY` function for that. Would simplify the logic greatly.
+
+## Rationale
+
+### Removing gas selectability
+
+One major change from the original `CALL` series of instructions is that the caller has no conrol over the amount of gas passed in as part of the call. The number of cases where such a feature is essential are probably better served by direct protocol integration.
+
+Removing gas selectability also introduces a valuable property that future revisions to the gas schedule will benefit from: you can always overcome Out of Gas errors by sending more gas as part of the transaction (subject to the block gas limit). Previously when raising storage costs ([EIP-1884](./eip-1884.md)) some contracts that sent only a limited amount of gas to their calls were broken by the new costing.
```

Comment:
Since you use OOG as an abbreviation later, you should expand it at least once beforehand:

```suggestion
Removing gas selectability also introduces a valuable property that future revisions to the gas schedule will benefit from: you can always overcome Out of Gas (OOG) errors by sending more gas as part of the transaction (subject to the block gas limit). Previously when raising storage costs ([EIP-1884](./eip-1884.md)) some contracts that sent only a limited amount of gas to their calls were broken by the new costing.
```